src/thunderbots/software/geom/util.h

```diff
@@ -486,3 +511,54 @@ Point getPointsMean(const std::vector<Point> &points);
  * @return the variance of the list of points
  */
 double getPointsVariance(const std::vector<Point> &points);
+
+
+/**
+ * Function returns the segment defined by the segment between the intersections of two
+ * Rays on a segment
+ *
+ * @param ray1 (Starting point and direction)
+ * @param ray2 (Starting point and direction)
+ * @param segment (Segment to find segment of intersection upon)
+ * @return Segment, the segment defined by the space between two intersecting rays on the
+ * segment parameter std::nullopt, if both rays don't intersect the segment, and the
+ * segment is not enclosed between the rays Segment, if one ray intersects the segment,
+ * but one of the segment parameters extremes are enclosed within the two rays
+ */
+std::optional<Segment> getIntersectingSegment(Ray ray1, Ray ray2, Segment segment);
+
+/**
+ * Function calculates whether the segment parameter is enclosed between the ray
+ * parameters. This means the entirety of the segment lays between the rays
+ *
+ * @param segment : segment parameter to calculate if its definition lies between the rays
+ * @param ray1 : Starting point and direction
+ * @param ray2 : Starting point an direction
+ * @return Segment: Returns the segment parameter if it is enclosed between ray1 and ray2
+ *         Returns std::nullopt of the ray is not completely enclosed between the rays, or
+ * not at all
+ */
+std::optional<Segment> segmentEnclosedBetweenRays(Segment segment, Ray ray1, Ray ray2);
+
+/**
+ * Function merges overlapping parallel segments into one combined segment
```

Comment:
Just to be clear, this includes partially overlapping segments right?

Reply:
Yup, it merged completely/partially overlapping segments into one segment that covers the same space as the two